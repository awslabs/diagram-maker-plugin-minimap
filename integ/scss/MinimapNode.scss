.example-minimap-node {
    box-sizing: border-box;
    border: 1px solid #545b64;
    background: #ffffff;
    padding: 0px;
    height: 100%;
    transition: all .2s ease;

    &.odd {
      border: 1px solid red;
    }

    &.even {
      border: 1px solid blue;
    }

    &.selected {
      border: 1px solid #0073bb;;
    }

    &.rectangle {
      padding: 1px;
    }

    &.circle {
      border-radius: 100%;
      padding: 0px;
      position: absolute;
      width: 100%;
      height: 100%;
      top: 0;
      left: 0;
    }
}
