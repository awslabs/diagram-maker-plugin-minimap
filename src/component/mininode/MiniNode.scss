.dm-mini-node{
    position: absolute;
    .dm-content {
      padding: 0px;
      position: relative;
      height: 100%;
      z-index: 1;
      box-sizing: border-box;
    }
  }
