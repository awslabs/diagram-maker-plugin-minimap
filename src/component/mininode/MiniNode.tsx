import { DestroyCallback, DiagramMakerNode } from 'diagram-maker';
import { default as ComposeView } from 'diagramMakerMinimap/component/common/ComposeView';
import { BoundRenderCallback } from 'diagramMakerMinimap/service';
import * as React from 'react';
import './MiniNode.scss';

export interface MiniNodeProps<NodeType> {
  renderCallback: BoundRenderCallback;
  destroyCallback: DestroyCallback;
  diagramMakerNode: DiagramMakerNode<NodeType>;
}

export default class MiniNode<NodeType> extends React.PureComponent<MiniNodeProps<NodeType>, {}> {
  public render(): JSX.Element {
    const { diagramMakerData } = this.props.diagramMakerNode;
    const { x, y } = diagramMakerData.position;
    const { width, height } = diagramMakerData.size;
    const transform = `translate3d(${x}px, ${y}px, 0)`;
    const { renderCallback, destroyCallback } = this.props;

    return (
      <div
        className="dm-mini-node"
        style={{ width, height, transform }}
      >
        <ComposeView
          renderCallback={renderCallback}
          destroyCallback={destroyCallback}
        />
      </div>
    );
  }
}
