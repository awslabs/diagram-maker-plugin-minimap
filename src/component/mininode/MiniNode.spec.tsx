import MiniNode from 'diagramMakerMinimap/component/mininode/MiniNode';
import * as React from 'react';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import './MiniNode.scss';

const destroyCallback = jest.fn();
const renderCallback = jest.fn();
const nodes = {
  diagramMakerData: {
    position: {
      x: 400,
      y: 600
    },
    size: {
      height: 300,
      width: 200
    }
  },
  id: 'myNode'
};
describe('MiniNode', () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('test MiniNode', () => {
    const renderer  = ShallowRenderer.createRenderer();

    renderer.render(
      <MiniNode
        diagramMakerNode={nodes}
        renderCallback={renderCallback}
        destroyCallback={destroyCallback}
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
