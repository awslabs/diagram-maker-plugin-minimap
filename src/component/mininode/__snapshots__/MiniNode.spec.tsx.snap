// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniNode test MiniNode 1`] = `
<div
  className="dm-mini-node"
  style={
    Object {
      "height": 300,
      "transform": "translate3d(400px, 600px, 0)",
      "width": 200,
    }
  }
>
  <ComposeView
    destroyCallback={[MockFunction]}
    renderCallback={[MockFunction]}
  />
</div>
`;
